### ISSUE TITLE
cli: move most client logic server-side

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#64643]
**tldr: we want the logic for many CLI commands to be moved to the server, and make the CLI command a mere "thin wrapper" that simply prints out what the server returns, without local processing.**

### Motivation

Today a lot of "useful" administrator logic is implemented via code running in a client command, that must be ran from a shell separately from a server.

For example: `cockroach node ls`, `cockroach node decommission`, `cockroach user-file` etc

The choice to implement this logic client-side makes it difficult to embed it inside a web page, or generally reuse it from other places in cockroachdb (unless the logic is entirely re-implemented).

Moreover, the programming idioms that we use in client code wrt error handling, SQL queries etc are different from those used in server code, which makes it difficult for engineers to produce code of equal quality on both sides.

### Strategy

What we wish instead is that each command, its parameters and its file inputs get sent to the server as an HTTP GET or POST request (see #51454 for context); have the server do all the processing and return a HTTP response, and have the client simply print out that response.

### Tactical steps

Given the large number of client commands, we will want to start with some "simpler" commands to demonstrate the concept and iron out how the APIs would look like. 

For example, the `cockroach node` sub-commands would be good candidates to start.

Then we can extend the logic to more complex commands, for example `cockroach userfile`.

**Comment by knz** [comment]
Relevant slack thread (internal): https://cockroachlabs.slack.com/archives/CA8H95NUU/p1620090115010300
